### ISSUE TITLE
False positive with `ConstructorInvokesOverridable`

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#658]
Consider this class:
```java
abstract class Test {
    private final Runnable run = () -> op();

    abstract void op();
}
```

Running with `-Xep:ConstructorInvokesOverridable:ERROR` yields:
```
Test.java:2: error: [ConstructorInvokesOverridable] Constructors should not invoke overridable methods.
    private final Runnable run = () -> op();
                                         ^
    (see http://errorprone.info/bugpattern/ConstructorInvokesOverridable)
1 error
```

Workarounds:
1. Replace the lambda expression with a method handle.
2. Replace the field with a method (even better).

Caveat: if the lambda expression were passed on to some other class while the object is still under construction, then `ConstructorInvokesOverridable` and/or [`ConstructorLeaksThis`](http://errorprone.info/bugpattern/ConstructorLeaksThis) _should_ flag that occurrence. (Because the code in that other class may execute the expression.)

**Comment by cushon** [comment]
98d25b520237aacb941bdf27ad4b73ffddacbb14
